core/common/src/test/java/alluxio/util/io/FileUtilsTest.java

```diff
@@ -256,4 +260,57 @@ public void createDirTest() throws IOException {
     Assert.assertTrue(FileUtils.exists(tempDir.getAbsolutePath()));
     Assert.assertTrue(tempDir.delete());
   }
+
+  /**
+   * Tests {@link FileUtils#createBlockPath} method when storage dir exists or not.
+   */
+  @Test
+  public void createStorageDirPathTest() throws IOException {
+    File storageDir = new File(mTestFolder.getRoot(), "tmp");
+    File blockFile = new File(storageDir, "200");
+
+    // When storage dir not exists
+    FileUtils.createBlockPath(blockFile.getAbsolutePath());
+    Assert.assertTrue(FileUtils.exists(storageDir.getAbsolutePath()));
+    Assert.assertEquals(
+        PosixFilePermissions.fromString("rwxrwxrwx"),
+        Files.getPosixFilePermissions(Paths.get(storageDir.getAbsolutePath())));
+
+    // When storage dir exists
+    FileUtils.createBlockPath(blockFile.getAbsolutePath());
+    Assert.assertTrue(FileUtils.exists(storageDir.getAbsolutePath()));
+  }
+
+  /**
+   * Tests invoking {@link FileUtils#createBlockPath} method concurrently. This simulates the case
+   * when multiple blocks belonging to the same storage dir get created concurrently.
+   */
+  @Test
+  public void concurrentCreateStorageDirPathTest() throws Exception {
+    class ConcurrentCreator implements Callable<Void> {
+      private final String mPath;
+
+      ConcurrentCreator(String path) {
+        mPath = path;
+      }
+
+      @Override
+      public Void call() throws Exception {
+        FileUtils.createBlockPath(mPath);
+        return null;
+      }
+    }
+
+    for (int i = 0; i < 100; i++) {
```

Comment:
If we want to really stress concurrency, it might help to run 10 threads at the same time and use a cyclic barrier or countdown latch to coordinate their calls to createBlockPath